tensorrt_llm/llmapi/mm_encoder.py

```diff
@@ -0,0 +1,163 @@
+from pathlib import Path
+from typing import Any, Optional, Union
+
+from tensorrt_llm.inputs.data import PromptInputs
+from .llm import BaseLLM, _TorchLLM, RequestOutput
+from tensorrt_llm.sampling_params import SamplingParams
+from tensorrt_llm._utils import nvtx_range_debug
+from typing import List, Sequence
+from tensorrt_llm.inputs import prompt_inputs
+from tqdm import tqdm
+from tensorrt_llm.inputs import create_input_processor
+from .llm_args import PybindMirror
+from .mpi_session import external_mpi_comm_available
+from tensorrt_llm.bindings import executor as tllm
+
+class MultimodalEncoder(_TorchLLM):
+    """MultimodalEncoder class is the main class for running a multimodal encoder model using PyTorch backend.
+"""
+
+    def __init__(self,
+                 model: Union[str, Path],
+                 trust_remote_code: bool = False,
+                 tensor_parallel_size: int = 1,
+                 dtype: str = "auto",
```

Comment:
Is it possible to have a `Literal` or a Enum here to describe the candidate set of the dtype argument? That will be helpful for other developers.

Reply:
Updated.